api/api.go

```diff
@@ -7,29 +7,38 @@ import (
 	"io/ioutil"
 	"net/http"
 
+	"github.com/ONSdigital/dp-authorisation/auth"
 	"github.com/ONSdigital/dp-image-api/apierrors"
+	"github.com/ONSdigital/dp-image-api/config"
 	"github.com/ONSdigital/log.go/log"
 	"github.com/gorilla/mux"
 )
 
 //API provides a struct to wrap the api around
 type API struct {
-	Router  *mux.Router
-	mongoDB MongoServer
+	Router      *mux.Router
+	mongoDB     MongoServer
+	permissions AuthHandler
```

Comment:
I just have a concern that the word `permissions` is nothing like `AuthHandler`. I might have been tempted to import the package as `dpauth` and call this `auth` so the line below would be…
```
… auth.Require(dpauth.Permissions{Create: true} …
```